server/bin/pbench-server-put-shim.py

```diff
@@ -46,18 +45,22 @@ def main(cfg_name: str) -> int:
     result_string = (
         f"{config.TS} Status: Total no. of tarballs {counts.ntotal},"
         f" Successfully moved {counts.ntbs}, Encountered"
-        f" {counts.ntotal-counts.ntbs} failures"
+        f" {counts.nerr} Errors"
     )
     logger.info(result_string)
 
     # prepare and send report
-    fd, tmp_file = tempfile.mkstemp(dir=config.TMP, text=True)
-    with os.fdopen(fd, "w+") as reportfp:
-        reportfp.write(f"{counts.nstatus}{result_string}\n")
+    try:
+        fd, tmp_file = tempfile.mkstemp(dir=config.TMP, text=True)
+    except Exception as exc:
+        logger.error("Creation of temporary Report file Failed: '{}'", exc)
```

Comment:
If the temporary file creation fails, then there is no point in proceeding with trying to create the report...we should just return, presumably with a status of `1`.

The next question is, if we get a non-zero error count at line 43, should we still exit with a status of `0` at line 70?  (Thoughts, @portante?)

Reply:
Yes, we should exit with `1` at line 70.